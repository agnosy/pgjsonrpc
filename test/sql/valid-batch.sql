\set ECHO none
BEGIN;
\t
\i sql/pgjsonrpc.sql

CREATE FUNCTION jsonrpc.subtract(p_request JSON)
RETURNS JSON AS
$function$
    SELECT jsonrpc.success_response(
        p_request,
        (
            (p_request->'params'->>0)::BIGINT -
            (p_request->'params'->>1)::BIGINT
        )
    );
$function$
LANGUAGE sql;

-- Create mapping from method echo to function_name
-- jsonrpc.echo
INSERT INTO jsonrpc.methods(name, description, function_name)
VALUES(
    'subtract',
    'Subtract the subtrahend from minuend passed in parameters.',
    'jsonrpc.subtract'
);

\set ECHO all

-- You should write your tests

SELECT jsonrpc.execute('
[
    {"jsonrpc": "2.0", "method": "sum", "params": [1,2,4], "id": "1"},
    {"jsonrpc": "2.0", "method": "subtract", "params": [42,23], "id": "2"},
    {"foo": "boo"},
    {"jsonrpc": "2.0", "method": "foo.get", "params": {"name": "myself"}, "id": "5"},
    {"jsonrpc": "2.0", "method": "get_data", "id": "9"} 
]
');

ROLLBACK;
