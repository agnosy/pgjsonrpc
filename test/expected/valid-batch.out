\set ECHO none
-- You should write your tests
SELECT jsonrpc.execute('
[
    {"jsonrpc": "2.0", "method": "sum", "params": [1,2,4], "id": "1"},
    {"jsonrpc": "2.0", "method": "subtract", "params": [42,23], "id": "2"},
    {"foo": "boo"},
    {"jsonrpc": "2.0", "method": "foo.get", "params": {"name": "myself"}, "id": "5"},
    {"jsonrpc": "2.0", "method": "get_data", "id": "9"} 
]
');
 [{"id": "1", "error": {"code": -32601, "message": "Method not found"}, "jsonrpc": "2.0"}, {"id": "2", "result": 19, "jsonrpc": "2.0"}, {"id": null, "error": {"code": -32600, "message": "Invalid Request"}, "jsonrpc": "2.0"}, {"id": "5", "error": {"code": -32601, "message": "Method not found"}, "jsonrpc": "2.0"}, {"id": "9", "error": {"code": -32601, "message": "Method not found"}, "jsonrpc": "2.0"}]

ROLLBACK;
